integration-tests/src/test/groovy/com/okta/sdk/tests/it/ApplicationsIT.groovy

```diff
@@ -517,4 +524,106 @@ class ApplicationsIT extends ITSupport {
+                  .setStateOrProvinceName("California")
+                  .setLocalityName("San Francisco")
+                  .setOrganizationName("Okta, Inc.")
+                  .setOrganizationalUnitName("Dev")
+                  .setCommonName("SP Issuer"))
+              .setSubjectAltNames(client.instantiate(CsrMetadataSubjectAltNames)
+                  .setDnsNames(["dev.okta.com"]))
+
+        // generate csr with metadata
+        Csr csr = app.generateCsr(csrMetadata)
+
+        // verify
+        assertPresent(app.listCsrs(), csr)
+
+        // revoke csr
+        app.revokeCsr(csr.getId())
+
+        // verify
+        assertNotPresent(app.listCsrs(), csr)
+    }
+
+    @Test
+    void oAuth2ScopeConsentGrantTest() {
+        Client client = getClient()
+
+        String label = "app-${uniqueTestName}"
+        Application app = client.instantiate(OpenIdConnectApplication)
+            .setLabel(label)
+            .setSettings(client.instantiate(OpenIdConnectApplicationSettings)
+                .setOAuthClient(client.instantiate(OpenIdConnectApplicationSettingsClient)
+                    .setClientUri("https://example.com/client")
+                    .setLogoUri("https://example.com/assets/images/logo-new.png")
+                    .setRedirectUris(["https://example.com/oauth2/callback",
+                                      "myapp://callback"])
+                    .setResponseTypes([OAuthResponseType.TOKEN,
+                                       OAuthResponseType.ID_TOKEN,
+                                       OAuthResponseType.CODE])
+                    .setGrantTypes([OAuthGrantType.IMPLICIT,
+                                    OAuthGrantType.AUTHORIZATION_CODE])
+                    .setApplicationType(OpenIdConnectApplicationType.NATIVE)
+                    .setTosUri("https://example.com/client/tos")
+                    .setPolicyUri("https://example.com/client/policy")))
+            .setCredentials(client.instantiate(OAuthApplicationCredentials)
+                .setOAuthClient(client.instantiate(ApplicationCredentialsOAuthClient)
+                    .setClientId(UUID.randomUUID().toString())
+                    .setAutoKeyRotation(true)
+                    .setTokenEndpointAuthMethod(OAuthEndpointAuthenticationMethod.CLIENT_SECRET_POST)))
+        client.createApplication(app)
+        registerForCleanup(app)
+
+        assertThat(app.getStatus(), equalTo(Application.StatusEnum.ACTIVE))
+
+        // grant consent
+        OAuth2ScopeConsentGrant oAuth2ScopeConsentGrant = app.grantConsentToScope(client.instantiate(OAuth2ScopeConsentGrant)
+            .setIssuer(client.dataStore.baseUrlResolver.baseUrl)
+            .setScopeId("okta.apps.manage"))
+
+        // verify
+        assertPresent(app.listScopeConsentGrants(), app.getScopeConsentGrant(oAuth2ScopeConsentGrant.getId()))
+
+        // revoke consent
+        app.revokeScopeConsentGrant(oAuth2ScopeConsentGrant.getId())
```

Comment:
Heads up that keys are not being removed, so you might see a "key storage limit" issue in the future. We ended up skipping all CSR/keys related ITs due to this issue.

Reply:
Noted.